cli/download_source.go

```diff
@@ -349,16 +358,41 @@ func cleanupTerraformFiles(path string, terragruntOptions *options.TerragruntOpt
 func getTerraformSourceUrl(terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) string {
 	if terragruntOptions.Source != "" {
 		return terragruntOptions.Source
-	} else if terragruntConfig.Terraform != nil {
-		return terragruntConfig.Terraform.Source
+	} else if terragruntConfig.Terraform != nil && terragruntConfig.Terraform.Source != nil {
+		return *terragruntConfig.Terraform.Source
 	} else {
 		return ""
 	}
 }
 
-// Download the code from the Canonical Source URL into the Download Folder using the terraform init command
-func terraformInit(terraformSource *TerraformSource, terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) error {
-	terragruntOptions.Logger.Printf("Downloading Terraform configurations from %s into %s using terraform init", terraformSource.CanonicalSourceURL, terraformSource.DownloadDir)
+// Download the code from the Canonical Source URL into the Download Folder using the go-getter library
+func downloadSource(terraformSource *TerraformSource, terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) error {
+	terragruntOptions.Logger.Printf("Downloading Terraform configurations from %s into %s", terraformSource.CanonicalSourceURL, terraformSource.DownloadDir)
+
+	// go-getter will not download into folders that already exist, so initially, we download into a temp folder
+	tmpDownloadDir, err := ioutil.TempDir("", "terragrunt-download-temp")
```

Comment:
Would it make sense to respect `TERRAGRUNT_DOWNLOAD_DIR` here? I'm thinking about issues on Windows with long filepaths, and a workaround of exporting something like `TERRAGRUNT_DOWNLOAD_DIR = "C:\.terragrunt-cache"`. If we first download instead to `TEMP`, that makes the workaround ineffective...

Reply:
Ah, that's a good point. Fixed in https://github.com/gruntwork-io/terragrunt/pull/731/commits/65741dbfe6cb36a843c054051e0056cb57767d36.